samples/nextjs/src/lib/page-props-factory.ts

```diff
@@ -42,11 +44,15 @@ const isServerSidePropsContext = function (
 };
 
 export class SitecorePagePropsFactory {
-  private componentPropsService: ComponentPropsService;
   private dictionaryService: DictionaryService;
 
-  constructor() {
-    this.componentPropsService = new ComponentPropsService();
+  constructor(
+    private componentPropsService: ComponentPropsService,
+    dictionaryServiceFactory: DictionaryServiceFactory,
+    private layoutPersonalizationUtils: LayoutPersonalizationUtils,
```

Comment:
Which you class are you referring as `this`: `LayoutPersonalizationUtils` or `SitecorePagePropsFactory`?

What is the benefit of having all methods static? Currently, we are using this service in unit tests and are able to mock\stub\spy them.

Reply:
I'm thinking that this `LayoutPersonalizationUtils` class not required to be a separate instance, can we have a separate functions, instead of defining class?

Reply:
The class gives a logical grouping of operations that are related to layout personalization. 
Could you please clarify what is the benefit of separate functions?